build/uploadperftobenchview.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
```

Comment:
Please run this through shellcheck (there is an online form via `https://www.shellcheck.net`)

This is reporting a number of items that should be fixed before merging:
```
$ shellcheck myscript
 
Line 53:
rm -r -f $perfWorkingDirectory/Microsoft.BenchView.JSONFormat > /dev/null 2>&1
         ^-- SC2154: perfWorkingDirectory is referenced but not assigned.
         ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 55:
if [ ! -d $perfWorkingDirectory/Microsoft.BenchView.JSONFormat ]; then
          ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 58:
    curl -o $perfWorkingDirectory/benchview.zip http://benchviewtestfeed.azurewebsites.net/api/v2/package/microsoft.benchview.jsonformat/0.1.0-pre024
            ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 59:
    unzip -q -o $perfWorkingDirectory/benchview.zip -d $perfWorkingDirectory/Microsoft.BenchView.JSONFormat
                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 71:
benchViewName="SDK perf $OS $architecture $configuration $runType $GIT_BRANCH_WITHOUT_ORIGIN"
                            ^-- SC2154: architecture is referenced but not assigned.
                                          ^-- SC2154: configuration is referenced but not assigned.
                                                         ^-- SC2154: runType is referenced but not assigned.
 
Line 74:
    benchViewName="$benchViewName $BenchviewCommitName"
                                  ^-- SC2154: BenchviewCommitName is referenced but not assigned.
 
Line 80:
echo BenchViewName: $benchViewName
                    ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 82:
echo Creating: $perfWorkingDirectory/submission-metadata.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 84:
                    -o "$perfWorkingDirectory/submission-metadata.json" || { echo Failed to create: $perfWorkingDirectory/submission-metadata.json && exit 1 ; }
>>                                                                                                  ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 86:
echo Creating: $perfWorkingDirectory/build.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 87:
python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/build.py" git --branch $GIT_BRANCH_WITHOUT_ORIGIN --type "$runType" \
>>                                                                                           ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 88:
                   -o "$perfWorkingDirectory/build.json" || { echo Failed to create: $perfWorkingDirectory/build.json && exit 1 ; }
>>                                                                                   ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 90:
echo Creating: $perfWorkingDirectory/machinedata.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 92:
                   -o "$perfWorkingDirectory/machinedata.json" || { echo Failed to create: $perfWorkingDirectory/machinedata.json && exit 1 ; }
>>                                                                                         ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 94:
echo Creating: $perfWorkingDirectory/measurement.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 95:
find $perfWorkingDirectory -maxdepth 1 -type f -name "*.xml" -exec echo Processing: "{}" \;
     ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 96:
find $perfWorkingDirectory -maxdepth 1 -type f -name "*.xml" -exec python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/measurement.py" xunitscenario "{}" --better desc --append \
     ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 99:
echo Creating: $perfWorkingDirectory/submission.json
               ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 112:
echo Uploading: $perfWorkingDirectory/submission.json
                ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 113:
python3.5 "$perfWorkingDirectory/Microsoft.BenchView.JSONFormat/tools/upload.py" "$perfWorkingDirectory/submission.json" --container coreclr || { echo Failed to upload: $perfWorkingDirectory/submission.json && exit 1 ; }
>>                                                                                                                                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.

$ 
```

Reply:
done...